source/appModules/notepad.py

```diff
@@ -10,11 +10,27 @@
 from typing import Optional
 import appModuleHandler
 import api
+import braille
+import controlTypes
+import eventHandler
+import speech
 import NVDAObjects
 
 
 class AppModule(appModuleHandler.AppModule):
 
+	def event_UIA_elementSelected(self, obj, nextHandler):
+		# Announce currently selected tab when it changes.
+		if (
+			obj.role == controlTypes.Role.TAB
+			# this is done because 2 selection events are sent for the same object, so to prevent double speaking.
+			and not eventHandler.isPendingEvents("UIA_elementSelected") and controlTypes.State.SELECTED in obj.states):
```

Comment:
Some formatting suggestions for readibility

```suggestion
			and not eventHandler.isPendingEvents("UIA_elementSelected") and controlTypes.State.SELECTED in obj.states
		):
```